util/servermemorylimit/servermemorylimit.go

```diff
@@ -81,12 +83,20 @@ type sessionToBeKilled struct {
 	sqlStartTime   time.Time
 	sessionID      uint64
 	sessionTracker *memory.Tracker
+
+	lastLogTime time.Time
 }
 
 func killSessIfNeeded(s *sessionToBeKilled, bt uint64, sm util.SessionManager) {
 	if s.isKilling {
 		if info, ok := sm.GetProcessInfo(s.sessionID); ok {
 			if info.Time == s.sqlStartTime {
+				if time.Since(s.lastLogTime) > 5*time.Second {
+					logutil.BgLogger().Warn("Memory usage Top1 SQL can't handle the kill signal",
+						zap.String("sql_text", info.Info),
+						zap.Int64("sql_memory_usage", info.MemTracker.BytesConsumed()))
```

Comment:
```suggestion
				if time.Since(s.lastLogTime) > 5*time.Second {
					logutil.BgLogger().Warn("Memory usage Top1 SQL can't handle the kill signal",
						zap.String("sql text", info.Info),
						zap.Int64("sql memory usage", info.MemTracker.BytesConsumed()))
```